### ISSUE TITLE
SMCC v2.2.1.2 not instance arm_uc_blockdevice

### ISSUE THREAD

**Issue by cyliangtw** [ARMmbed/pelion-ready-example#64]
In SMC v2.2.1.1, this pelion-ready-sample could link symbols well. In `simple-mbed-cloud-client.cpp`, it will instance `arm_uc_blockdevice`.
However, got linking failure after apply the last update with SMC v2.2.1.2, details of error message as:
```
[ERROR] Error: L6218E: Undefined symbol arm_uc_blockdevice (referred from BUILD/NUMAKER_IOT_M487/ARM/simple-mbed-cloud-client/mbed-cloud-client/update-client-hub/modules/pal-blockdevice/source/arm_uc_pal_blockdevice_mbed.o).
Finished: 0 information, 0 warning and 1 error messages.
```

**Comment by screamerbg** [comment]
@cyliangtw Could you please share more information about your setup and environment? Is this while trying to compile tests, or the main application?

**Comment by cyliangtw** [comment]
Just for main application demo code.


===========================================================================================
The privileged confidential information contained in this email is intended for use only by the addressees as indicated by the original sender of this email. If you are not the addressee indicated in this email or are not responsible for delivery of the email to such a person, please kindly reply to the sender indicating this fact and delete all copies of it from your computer and network server immediately. Your cooperation is highly appreciated. It is advised that any unauthorized use of confidential information of Nuvoton is strictly prohibited; and any information in this email irrelevant to the official business of Nuvoton shall be deemed as neither given nor endorsed by Nuvoton.

**Comment by cyliangtw** [comment]
@screamerbg , clone the master branch, smcc v2.2.1.2 + OS 5.11.5, revise "use-button" as false, then `$ mbed compile -m NUMAKER_IOT_M487 -t ARMC6`   got error message:
```
Error: L6218E: Undefined symbol arm_uc_blockdevice (referred from BUILD/NUMAKER_IOT_M487/ARMC6/simple-mbed-cloud-client/mbed-cloud-client/update-client-hub/modules/pal-blockdevice/source/arm_uc_pal_blockdevice_mbed.o).
Finished: 0 information, 5 warning and 1 error messages.

```

**Comment by cyliangtw** [comment]
@screamerbg, resolved after applied mbed-cloud-client 2.2.1 instead of 2.0.1 .
